js/itbit.js

```diff
@@ -121,19 +121,54 @@ module.exports = class itbit extends Exchange {
 
     parseTrade (trade, market) {
         let timestamp = this.parse8601 (trade['timestamp']);
-        let id = trade['matchNumber'].toString ();
-        return {
-            'info': trade,
-            'timestamp': timestamp,
-            'datetime': this.iso8601 (timestamp),
-            'symbol': market['symbol'],
-            'id': id,
-            'order': id,
-            'type': undefined,
-            'side': undefined,
-            'price': this.safeFloat (trade, 'price'),
-            'amount': this.safeFloat (trade, 'amount'),
-        };
+        if ('matchNumber' in trade) {
+            // then this is a public request response, ie, market data
+            let id = trade['matchNumber'].toString ();
+            return {
+                'info': trade,
+                'timestamp': timestamp,
+                'datetime': this.iso8601 (timestamp),
+                'symbol': market['symbol'],
+                'id': id,
+                'order': id,
+                'type': undefined,
+                'side': undefined,
+                'price': this.safeFloat (trade, 'price'),
+                'amount': this.safeFloat (trade, 'amount'),
+            };
+        } else {
+            // this is a private request response, ie, MY trades
+            // note on fee from itbit:
+            // + **commissionPaid** `string` - *net trade fee paid after using any available rebate balance*
+            let fee = {
+                'cost': this.safeFloat (trade, 'commissionPaid'),
+                'currency': this.safeString (trade, 'commissionCurrency'),
```

Comment:
We should not ignore `commonCurrencies` here, otherwise it can cause inconsistencies if the user has defined an override for a symbol or a currency code in the options.